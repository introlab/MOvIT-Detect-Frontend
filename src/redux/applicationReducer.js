/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

// ------------------------------------
// Constants
// ------------------------------------

let url;
const isDemo = false;
const offset = 0;// Math.ceil(new Date().getTimezoneOffset() / 60) * -1;

const bport = process.env.BPORT || 1880; // Backend port
const bhost = process.env.BHOST || '192.168.10.1'; // Host address for the backend

// url = `http://${bhost}:${bport}/`; // URL for the backend, this URL is exported and used throughout the code

// DL changed URL to to relative path to be proxied internally
url = '/api/';

console.log(`Mode: ${process.env.NODE_ENV}, Backend URL: ${url}`);

export const URL = url;
export const OFFSET = offset;
export const IS_DEMO = isDemo;
export const IS_MOBILE = window.innerWidth <= 500;
export const IS_TABLET = window.innerWidth <= 780;
export const LANGUAGE = 'LANGUAGE';
export const LANGUAGE_UPDATE = 'LANGUAGE_UPDATE';
export const FR = 'FR';
export const EN = 'EN';
export const cFR = 'cFR';
export const cEN = 'cEN';
export const PROFILE = 'PROFILE';
export const TOKEN = 'TOKEN';

// ------------------------------------
// Actions
// ------------------------------------
function changeLanguage() {
  return {
    type: LANGUAGE,
  };
}

function updateLanguage() {
  return {
    type: LANGUAGE_UPDATE,
  };
}

function changeProfile(profileName) {
  return {
    type: PROFILE,
    profile: profileName,
  };
}

function changeToken(tokenString) {
  return {
    type: TOKEN,
    token: tokenString,
  };
}


// Update the ui language when exitting clinician view
function updateUILanguage(lan, user) {
  let toReturn = FR;
  if (user === 'clinician') {
    if (lan.includes('FR')) {
      toReturn = cFR;
    } else {
      toReturn = cEN;
    }
  } else if (lan.includes('FR')) {
    toReturn = FR;
  } else {
    toReturn = EN;
  }
  console.log(toReturn);
  return toReturn;
}


// Toggle the language between EN and FR for the user and cFR and cEN for the clinician
function switchLanguage(lan, user) {
  let toReturn = FR;
  if (user === 'clinician') {
    if (lan.includes('FR')) {
      toReturn = cFR;
    } else {
      toReturn = cEN;
    }
  } else if (lan.includes('FR')) {
    toReturn = FR;
  } else {
    toReturn = EN;
  }

  if (toReturn === FR) {
    toReturn = EN;
  } else if (toReturn === EN) {
    toReturn = FR;
  } else if (toReturn === cEN) {
    toReturn = cFR;
  } else if (toReturn === cFR) {
    toReturn = cEN;
  }
  console.log(toReturn);
  return toReturn;
}

export const ApplicationActions = {
  changeLanguage,
  changeProfile,
  changeToken,
  updateLanguage,
};
// ------------------------------------
// Action Handlers
// ------------------------------------
const ACTION_HANDLERS = {
  [LANGUAGE]: state => (
    { ...state, language: switchLanguage(state.language, state.profile) }
  ),
  [LANGUAGE_UPDATE]: state => (
    { ...state, language: updateUILanguage(state.language, state.profile) }
  ),
  [PROFILE]: (state, action) => (
    { ...state, profile: action.profile }
  ),
  [TOKEN]: (state, action) => (
    { ...state, token: action.token, header: { headers: { Authorization: action.token } } }
  ),
};

// ------------------------------------
// Reducer
// ------------------------------------

export const initApplication = {
  language: 'FR',
  profile: '',
  userName: '',
  userID: '',
  token: null,
  header: {},
  maxAngle: null,
  userWeight: null,
};
export default function applicationReducer(state = initApplication, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
