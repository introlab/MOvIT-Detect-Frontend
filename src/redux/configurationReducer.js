/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin Didier Tran
 */

export const USER_NAME = 'USER_NAME';
export const USER_ID = 'USER_ID';
export const USER_WEIGHT = 'USER_WEIGHT';
export const MAX_ANGLE = 'MAX_ANGLE';
export const TELASK_KEY = 'TELASK_KEY';
export const TELASK_USERNAME = 'TELASK_USERNAME';
export const TELASK_HOST = 'TELASK_HOST';


// ------------------------------------
// Actions
// ------------------------------------

function changeUserName(name) {
  return {
    type: USER_NAME,
    userName: name,
  };
}

function changeTelaskKey(key) {
  return {
    type: TELASK_KEY,
    telaskKey: key,
  };
}

function changeTelaskUsername(username) {
  return {
    type: TELASK_USERNAME,
    telaskUsername: username,
  };
}

function changeTelaskHost(host) {
  return {
    type: TELASK_HOST,
    telaskHost: host,
  };
}

function changeUserID(id) {
  return {
    type: USER_ID,
    userID: id,
  };
}
function changeUserWeight(weight) {
  return {
    type: USER_WEIGHT,
    userWeight: weight,
  };
}
function changeMaxAngle(angle) {
  return {
    type: MAX_ANGLE,
    maxAngle: angle,
  };
}
export const ConfigurationActions = {
  changeUserName,
  changeUserID,
  changeUserWeight,
  changeMaxAngle,
  changeTelaskKey,
  changeTelaskHost,
  changeTelaskUsername,
};

const ACTION_HANDLERS = {
  [USER_NAME]: (state, action) => (
    { ...state, userName: action.userName }
  ),
  [USER_ID]: (state, action) => (
    { ...state, userID: action.userID }
  ),
  [USER_WEIGHT]: (state, action) => (
    { ...state, userWeight: action.userWeight }
  ),
  [MAX_ANGLE]: (state, action) => (
    { ...state, maxAngle: action.maxAngle }
  ),
  [TELASK_KEY]: (state, action) => (
    { ...state, telaskKey: action.telaskKey }
  ),
  [TELASK_HOST]: (state, action) => (
    { ...state, telaskHost: action.telaskHost }
  ),
  [TELASK_USERNAME]: (state, action) => (
    { ...state, telaskUsername: action.telaskUsername}
  ),
};

// ------------------------------------
// Reducer
// ------------------------------------

export const initConfiguration = {
  userName: '',
  userID: '',
  maxAngle: null,
  userWeight: null,
  telaskKey: '',
  telaskUsername: '',
  telaskHost: '',
};
export default function applicationReducer(state = initConfiguration, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
