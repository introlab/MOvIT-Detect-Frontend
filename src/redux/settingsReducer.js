import '../styles/components/moduleStatus.css';

/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

export const DATA_AGREEMENT = 'DATA_AGREEMENT';
export const SNOOZE_TIME = 'SNOOZE_TIME';
export const ENABLED = 'ENABLED';
export const IS_LED_BLINKING_ENABLED = 'IS_LED_BLINKING_ENABLED';
export const IS_VIBRATION_ENABLED = 'IS_VIBRATION_ENABLED';
export const TOTAL_MEMORY = 'TOTAL_MEMORY';
export const USED_MEMORY = 'USED_MEMORY';
export const MODULES_STATUS = 'MODULES_STATUS';
export const IS_UPDATE_AVAILABLE = 'IS_UPDATE_AVAILABLE';
export const LAST_UPDATE_DATE = 'LAST_UPDATE_DATE';
export const WIFI_CONNECTION = 'WIFI_CONNECTION';

// -------------- NOTIFICATION SETTINGS --------------
function changeAreNotificationsEnabled(isEnabled) {
  return {
    type: ENABLED,
    enabled: isEnabled,
  };
}

function changeIsLedBlinkingEnabled(isEnabled) {
  return {
    type: IS_LED_BLINKING_ENABLED,
    isLedBlinkingEnabled: isEnabled,
  };
}

function changeIsVibrationEnabled(isEnabled) {
  return {
    type: IS_VIBRATION_ENABLED,
    isVibrationEnabled: isEnabled,
  };
}

function changeSnoozeTime(time) {
  return {
    type: SNOOZE_TIME,
    snoozeTime: time,
  };
}

// -------------- MEMORY --------------
function changeTotalMemory(totalMemory) {
  return {
    type: TOTAL_MEMORY,
    totalMemory,
  };
}

function changeUsedMemory(usedMemory) {
  return {
    type: USED_MEMORY,
    usedMemory,
  };
}

// -------------- PERMISSIONS --------------
function changeDataAgreement(agree) {
  return {
    type: DATA_AGREEMENT,
    dataAgreement: agree,
  };
}

// -------------- MODULES STATUS --------------
function changeModulesStatus(modulesStatus) {
  return {
    type: MODULES_STATUS,
    modulesStatus,
  };
}

function changeIsUpdateAvailable(isAvailable) {
  return {
    type: IS_UPDATE_AVAILABLE,
    isUpdateAvailable: isAvailable,
  };
}

// -------------- WIFI --------------
function changeIsWifiConnected(isConnected) {
  return {
    type: WIFI_CONNECTION,
    isWifiConnected: isConnected,
  };
}

export const SettingsActions = {
  changeAreNotificationsEnabled,
  changeIsLedBlinkingEnabled,
  changeIsVibrationEnabled,
  changeSnoozeTime,
  changeTotalMemory,
  changeUsedMemory,
  changeDataAgreement,
  changeModulesStatus,
  changeIsUpdateAvailable,
  changeIsWifiConnected,
};

const ACTION_HANDLERS = {
  [DATA_AGREEMENT]: (state, action) => (
    { ...state, dataAgreement: action.dataAgreement }
  ),
  [ENABLED]: (state, action) => (
    { ...state, enabled: action.enabled }
  ),
  [IS_LED_BLINKING_ENABLED]: (state, action) => (
    { ...state, isLedBlinkingEnabled: action.isLedBlinkingEnabled }
  ),
  [IS_VIBRATION_ENABLED]: (state, action) => (
    { ...state, isVibrationEnabled: action.isVibrationEnabled }
  ),
  [SNOOZE_TIME]: (state, action) => (
    { ...state, snoozeTime: action.snoozeTime }
  ),
  [TOTAL_MEMORY]: (state, action) => (
    { ...state, totalMemory: action.totalMemory }
  ),
  [USED_MEMORY]: (state, action) => (
    { ...state, usedMemory: action.usedMemory }
  ),
  [MODULES_STATUS]: (state, action) => (
    { ...state, modulesStatus: action.modulesStatus }
  ),
  [IS_UPDATE_AVAILABLE]: (state, action) => (
    { ...state, isUpdateAvailable: action.isUpdateAvailable }
  ),
  [WIFI_CONNECTION]: (state, action) => (
    { ...state, isWifiConnected: action.isWifiConnected }
  ),
};

export const initParameter = {
  dataAgreement: true,
  enabled: true,
  isLedBlinkingEnabled: true,
  isVibrationEnabled: true,
  isUpdateAvailable: false,
  isWifiConnected: false,
  snoozeTime: 10,
  totalMemory: 0,
  usedMemory: 0,
  modulesStatus: {},
};

export default function applicationReducer(state = initParameter, action) {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
}
