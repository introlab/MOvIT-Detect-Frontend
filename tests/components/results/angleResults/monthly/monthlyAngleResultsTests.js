/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import MockAdapter from 'axios-mock-adapter';
import PropTypes from 'prop-types';
import React from 'react';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import sinon from 'sinon';
import axios from 'axios';
import MonthlyAngleResults from '../../../../../src/components/results/angleResults/monthly/monthlyAngleResults';
import { OFFSET, URL } from '../../../../../src/redux/applicationReducer';


Enzyme.configure({ adapter: new Adapter() });

const month = '1';

describe('MonthlyAngleResults Tests', () => {
  let wrapper;

  const initialState = {
    applicationReducer: { language: 'en' },
    recommendationReducer: { reduceSlidingRest: true },
  };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    month,
  };

  const response = {
    1: [
      0.7,
      0.20,
    ],
    2: [
      0.45,
      0.25,
    ],
  };

  function initializeMockAdapter() {
    const mock = new MockAdapter(axios);
    const date = new Date(new Date().getFullYear(), month, 1);

    mock.onGet(`${URL}monthlySlideProgress?Day=${+date}&Offset=${OFFSET}`).reply(200, response);
  }

  initializeMockAdapter();

  beforeEach(() => {
    wrapper = shallow(<MonthlyAngleResults store={store} {...props} />).dive();
  });

  it('should have proptypes', () => {
    const actualValue = MonthlyAngleResults.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      month: PropTypes.string.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should update month', () => {
    wrapper.instance().componentWillReceiveProps({ month: '4' });

    expect(wrapper.state('month')).toEqual('4');
  });

  it('should get the month data when receiving new props', () => {
    const spy = sinon.spy(wrapper.instance(), 'getMonthlySlidingProgress');

    wrapper.setProps({ month: '2' });

    expect(wrapper.state('month')).toEqual('2');
    expect(spy.calledOnce).toEqual(true);
  });

  it('should do nothing when receiving matching props', () => {
    const spy = sinon.spy(wrapper.instance(), 'getMonthlySlidingProgress');

    wrapper.setProps({ month });

    expect(wrapper.state('month')).toEqual(month);
    expect(spy.calledOnce).toEqual(false);
  });

  it('should get the month data', async () => {
    const spy = sinon.spy(wrapper.instance(), 'loadMonthlySlidingData');

    await wrapper.instance().getMonthlySlidingProgress(month);

    expect(spy.getCalls()[0].args[0]).toEqual(response);
    expect(wrapper.state('isLoaded')).toEqual(true);
    expect(wrapper.state('hasErrors')).toEqual(false);
    expect(wrapper.state('monthSlideLabels')).toEqual(['1', '2']);
    expect(wrapper.state('monthSildeRest')).toEqual([response[1][0] * 100, response[2][0] * 100]);
    expect(wrapper.state('monthSildeMoving')).toEqual([response[1][1] * 100, response[2][1] * 100]);
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
