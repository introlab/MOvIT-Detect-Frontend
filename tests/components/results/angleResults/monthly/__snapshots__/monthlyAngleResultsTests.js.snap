// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyAngleResults Tests should match the snapshot 1`] = `
<div>
  <div
    className="col-lg-2 leftMenu"
  >
    <ul
      className="graphlist"
    >
      <li
        className="graphLink"
      >
        <a
          href="results/angle#monthlyAngle"
        >
          results.graphicsLink.angle.en
        </a>
      </li>
      <li
        className="graphLink"
      >
        <a
          href="results/angle#monthlyTilt"
        >
          SuccessfulTilt.tiltMade.en
        </a>
      </li>
      <li
        className="graphLink"
      >
        <a
          href="results/angle#reduceSlidingRest"
        >
          monthlyResults.rest.en
        </a>
      </li>
    </ul>
  </div>
  <div
    className=" col-lg-10 offset-lg-2 results resultsContainer"
  >
    <div
      className="col-lg-8 offset-lg-1"
    >
      <div>
        <div>
          <Connect(MonthlyAngleDistribution)
            month="1"
          />
          <Connect(MonthlySuccessTilt)
            month="1"
          />
          <div>
            <div
              id="reduceSlidingMoving"
            >
              <GoalChart
                data={
                  Object {
                    "datasets": Array [
                      Object {
                        "borderColor": "red",
                        "data": Array [],
                        "fill": true,
                        "label": "monthlyResults.travel.successRate.en",
                        "lineTension": 0,
                      },
                    ],
                    "labels": Array [],
                  }
                }
                options={
                  Object {
                    "legend": Object {
                      "onClick": null,
                    },
                    "scales": Object {
                      "yAxes": Array [
                        Object {
                          "ticks": Object {
                            "callback": [Function],
                            "max": 100,
                            "min": 0,
                          },
                        },
                      ],
                    },
                    "tooltips": Object {
                      "callbacks": Object {
                        "label": [Function],
                      },
                    },
                  }
                }
                successMessage="monthlyResults.travel.success.en"
                title="monthlyResults.travel.en"
              />
            </div>
            <div
              id="reduceSlidingRest"
            >
              <GoalChart
                condition={true}
                data={
                  Object {
                    "datasets": Array [
                      Object {
                        "borderColor": "red",
                        "data": Array [],
                        "fill": true,
                        "label": "monthlyResults.travel.successRate.en",
                        "lineTension": 0,
                      },
                    ],
                    "labels": Array [],
                  }
                }
                options={
                  Object {
                    "legend": Object {
                      "onClick": null,
                    },
                    "scales": Object {
                      "yAxes": Array [
                        Object {
                          "ticks": Object {
                            "callback": [Function],
                            "max": 100,
                            "min": 0,
                          },
                        },
                      ],
                    },
                    "tooltips": Object {
                      "callbacks": Object {
                        "label": [Function],
                      },
                    },
                  }
                }
                successMessage="monthlyResults.rest.success.en"
                title="monthlyResults.rest.en"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
